Datagrid/ProxyQuery.php

```diff
@@ -309,24 +309,16 @@ protected function getFixedQueryBuilder(QueryBuilder $queryBuilder)
             }
             $idxSelect .= ($idxSelect !== '' ? ', ' : '').$idSelect;
         }
-        $queryBuilderId->resetDQLPart('select');
-        $queryBuilderId->add('select', 'DISTINCT '.$idxSelect);
+        $queryBuilderId->select($idxSelect);
+        $queryBuilderId->distinct();
 
         // for SELECT DISTINCT, ORDER BY expressions must appear in idxSelect list
         /* Consider
             SELECT DISTINCT x FROM tab ORDER BY y;
         For any particular x-value in the table there might be many different y
         values.  Which one will you use to sort that x-value in the output?
         */
-        // todo : check how doctrine behave, potential SQL injection here ...
-        if ($this->getSortBy()) {
-            $sortBy = $this->getSortBy();
-            if (strpos($sortBy, '.') === false) { // add the current alias
-                $sortBy = $rootAlias.'.'.$sortBy;
-            }
-            $sortBy .= ' AS __order_by';
-            $queryBuilderId->addSelect($sortBy);
-        }
```

Comment:
Column from `getSortBy` is already added to `orderBy` list in `execute` and will be handled by `addOrderedColumns` below.